wemake_python_styleguide/visitors/ast/operators.py

```diff
@@ -268,3 +269,35 @@ def visit_NamedExpr(
         """
         self.add_violation(consistency.WalrusViolation(node))
         self.generic_visit(node)
+
+
+@final
+class BitwiseOpVisitor(base.BaseNodeVisitor):
+    """Checks bitwise operations are used correctly."""
+
+    def visit_BinOp(self, node: ast.BinOp) -> None:
+        """
+        Finds bad usage of bitwise operation with binary operation.
+
+        Raises:
+            BitwiseAndBooleanMixupViolation
+
+        """
+        self._check_BitOp(node)
+        self.generic_visit(node)
+
+    def _check_BitOp(self, node: ast.BinOp) -> None:
+        if not isinstance(node.op, (ast.BitOr, ast.BitAnd)):
+            return
+
+        if (self._check_sides(node.left) or self._check_sides(node.right)):
+            self.add_violation(BitwiseAndBooleanMixupViolation(node))
+
+    # checks either side of the Bitwise operation invalid usage
+    def _check_sides(self, node) -> bool:
+        invalid = False
```

Comment:
We can refactor this into:

`return isinstance(node, self._invalid_nodes)`

Where `self._invalid_nodes` is a class-level constant with all the nodes we need. Check other visitor for examples, search `AnyNodes` type.

Reply:
Thanks for this suggestion, I was feeling that my code there was a little inelegant.